service/frontend/nexus_http_handler.go

```diff
@@ -160,6 +156,103 @@ func (h *NexusHTTPHandler) dispatchNexusTaskByNamespaceAndTaskQueue(w http.Respo
 		h.writeNexusFailure(w, http.StatusBadRequest, &nexus.Failure{Message: err.Error()})
 		return
 	}
+
+	r, err = h.parseTlsAndAuthInfo(r, &nc)
+	if err != nil {
+		h.logger.Error("failed to get claims", tag.Error(err))
+		h.writeNexusFailure(w, http.StatusUnauthorized, &nexus.Failure{Message: "unauthorized"})
+		return
+	}
+
+	u, err := mux.CurrentRoute(r).URL("namespace", params.Namespace, "task_queue", params.TaskQueue)
+	if err != nil {
+		h.logger.Error("invalid URL", tag.Error(err))
+		h.writeNexusFailure(w, http.StatusInternalServerError, &nexus.Failure{Message: "internal error"})
+		return
+	}
+
+	h.serveResolvedURL(w, r, u)
+}
+
+// Handler for [nexushttp.RouteSet.DispatchNexusTaskByService].
+func (h *NexusHTTPHandler) dispatchNexusTaskByService(w http.ResponseWriter, r *http.Request) {
+	if !h.enabled() {
+		h.writeNexusFailure(w, http.StatusNotFound, &nexus.Failure{Message: "nexus endpoints disabled"})
+		return
+	}
+
+	service := parseRequestParams(commonnexus.RouteDispatchNexusTaskByService, w, r)
+
+	serviceID, err := url.PathUnescape(*service)
+	if err != nil {
+		h.logger.Error("invalid URL", tag.Error(err))
+		h.writeNexusFailure(w, http.StatusBadRequest, &nexus.Failure{Message: "invalid URL"})
+		return
+	}
+	serviceInfo, err := h.incomingServiceRegistry.Get(r.Context(), serviceID)
+	if err != nil {
+		h.logger.Error("invalid Nexus incoming service ID", tag.Error(err))
+
+		var notFound *serviceerror.NotFound
+		switch {
+		case errors.As(err, &notFound):
+			h.writeNexusFailure(w, http.StatusNotFound, &nexus.Failure{Message: err.Error()})
+		case errors.Is(err, context.DeadlineExceeded):
+			h.writeNexusFailure(w, http.StatusRequestTimeout, &nexus.Failure{Message: err.Error()})
+		default:
+			h.writeNexusFailure(w, http.StatusInternalServerError, &nexus.Failure{Message: err.Error()})
+		}
+		return
+	}
+
+	nc := h.getNexusContextFromService(serviceInfo)
+
+	r, err = h.parseTlsAndAuthInfo(r, &nc)
+	if err != nil {
+		h.logger.Error("failed to get claims", tag.Error(err))
+		h.writeNexusFailure(w, http.StatusUnauthorized, &nexus.Failure{Message: "unauthorized"})
+		return
+	}
+
+	u, err := mux.CurrentRoute(r).URL("service", *service)
+	if err != nil {
+		h.logger.Error("invalid URL", tag.Error(err))
+		h.writeNexusFailure(w, http.StatusInternalServerError, &nexus.Failure{Message: "internal error"})
+		return
+	}
+
+	h.serveResolvedURL(w, r, u)
+}
+
+func (h *NexusHTTPHandler) getBaseNexusContext(apiName string) nexusContext {
```

Comment:
nit: I may be taking this rule too far but I try to avoid using `get` in method names: https://go.dev/doc/effective_go#Getters